intros/TypeScript for the Haskell or ML Programmer.md

```diff
@@ -0,0 +1,658 @@
+# TypeScript for Functional Programmers
+
+TypeScript began its life as an attempt to bring traditional object-oriented types
+to JavaScript so that the programmers at Microsoft could bring
+traditional object-oriented programs to the web. As it has developed, TypeScript's type
+system has evolved to model code written by native JavaScripters. The
+resulting system is powerful, interesting and messy.
+
+This introduction is designed for working Haskell or ML programmers
```

Comment:
Maybe Lisp too?

Reply:
F# too

Reply:
I think F#, Caml and SML programmers all know that they're ML programmers, and I think the page title is good enough to get them here.

Lisp programmers are usually not considered functional programmers these days since Lisp doesn't have static typing, ADTs or pattern matching built in. The FP languages divide into statically typed and dynamically typed, but culturally, [Backus' 1977 Turing award lecture](https://www.thocp.net/biographies/papers/backus_turingaward_lecture.pdf) that led to the development of ML is the real dividing line.